### DISCUSSION TITLE
Vale a pena iniciar um novo projeto (do zero) com Angular 1.x?

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#403] [Discussão]
Fala galera!

Vocês acham que ainda vale a pena utilizar o Angular 1.x para COMEÇAR um novo projeto? Por que?

**Comment by leocavalcante** [comment]
É difícil responder sem a analise de requisitos desse projeto e sem conhecer o time que vai executar.

De forma genérica,  diria que não, porque é uma tecnologia que caminha para obsolescência, já não recebe mais features (só patches com fix de bug e segurança), a comunidade esta entendendo que será passado e esta migrando pro A2 ou até outros frameworks, não usa VirtualDOM etc. Mas essa é provavelmente a resposta que você já esperava ler, a de seguir o curso natural das coisas, evoluir junto e não ficar para trás.

Só que, IMHO, não é tão simples assim.

- A equipe já conhece Angular 2 ou outro framework?
- O projeto permite um tempo de aprendizagem/learning curve?
- O projeto permite ser uma espécie de "cobaia" pra algo novo na equipe ou precisa ser muito sólido?
- Algum requisito do projeto só é atendido pelo A1?
- A tree view será muito complexa/Performance no render é uma key no projeto?
- O projeto ficará por tempo indeterminado no ar ou é um app pra uma campanha de curto prazo, por exemplo?
- Etc...

**Comment by willycamargo** [comment]
Concordo com o que o @leocavalcante disse e...

IMHO, só valeria a pena se:
* A equipe que for trabalhar no projeto já tenha experiência com o Angular 1 e não tenha experiência com outros frameworks/libs.
* O projeto tem um prazo muito apertado, o que impossibilita a possibilidade de estudar Angular 2 ou outros frameworks/libs.

Ou seja:
* Não acho viável ter que ensinar e treinar alguém para trabalhar com Angular 1.
* Não acho viável utilizar Angular 1 quando se pode utilizar Angular 2 ou outros frameworks.

**Comment by woliveiras** [comment]
Legal galera! :metal: :metal: :metal:

É uma questão bem genérica mesmo.

Gostaria de saber justamente o que estão comentando: prós e contras de se continuar usando ele.

Abri outra issue que acho que vocês contribuiriam muito: #404

**Comment by Pompeu** [comment]
eu comecei um novo em 1.x em março do ano passado, me arrependo, esse ano ou é 2 ou react, minha motivação é que gosto de coisas novas aprender novidades e resolver problemas de forma mais simples.

**Comment by alexsandro-xpt** [comment]
Todos 2 lados tem um risco a seguir.

Mês passado um integrante do meu time adicionou A1 em um projeto que já existia.

**Comment by rodrigo-fonseca** [comment]
Começa com o 1.x de modo que seja fácil migrar para o Angular 2.x.

**Comment by brunoti** [comment]
@eullercristian Ionic 2 utiliza Angular 2 e está bem alinhado com os releases e tenho gostado muito de trabalhar com ele.

**Comment by sibelius** [comment]
react native é bem melhor do que o ionic

Vai de React (react minha escolha)
ou de angular 2

Temos a comunidade do react brasil para solucionar as dúvidas: http://react-brasil-slack.herokuapp.com/

**Comment by felipemouradev** [comment]
Se a equipe souber o angularjs 1.x e o prazo for curto vale a pena sim... Caso o projeto te permita aprender outra coisas, nem volta mais para o angularjs, pega outra coisa, aurelia, vue, etc... Na minha opinião jogaram a simplicidade do angularjs no lixo... muito código para fazer pouca coisa...

**Comment by williamokano** [comment]
Não

**Comment by yanmagale** [comment]
Sou da mesma opinião do que vocês falaram, depende muito do escopo do projeto, prazo e da equipe que irá se executar. Mas em termos de projetos mais duradouros, seria a V2 do Angular, ou outro framework a sua escolha.

O impacto da V1 do Angular é tão grande, que os projetos Ionic v1 estão sofrendo as mesmas consequências das aplicações V1. Nosso time esta numa grande discussão, porque qualquer tentativa de refactoring para upgrade do produto, irá nos levar a uma troca de stack. Então estamos vendo sobre React Native ou Weex.

**Comment by jeansarlon** [comment]
Eu compartilho da opinião do pessoal ai, iniciar um projeto em Angular 1 somente em casos bem específicos já citados!
Eu estou trabalhando atualmente com o VueJS e Quasar framework
http://quasar-framework.org/

**Comment by rubenmarcus** [comment]
Mas.. tem bastante vaga Angular 1.x ainda..

**Comment by deividmarques** [comment]
@woliveiras prefiro usar o vue, ou mesmo o próprio React, do que angular 1

**Comment by woliveiras** [comment]
O principal problema do Angular 1.x, sem levar em conta coisas como prazo do projeto e equipe seria, então, o suporte que vai acabar, certo galera?

**Comment by woliveiras** [comment]
Tem muita gente falando bem do Vuejs mesmo. Vou dar aquela conferida com carinho, viu. :grin:

**Comment by leandroruel** [comment]
comecei um projeto com o 1.x e quando vi que ia lançar o 2, que já lançou já... abandonei o projeto com angular 1.x e deixei num branch separado guardado, agora tô vendo como vou fazer com React.

**Comment by AbraaoAlves** [comment]
Ainda vale **SIM**. 
(E não, não sou defensor do AngularJS, nem Angular2, sou um critico bem azedo na verdade. Já o usei  bastante(+2 anos) para não querer mais em meus projetos pessoais.)

#### _"Mas porque SIM?"_

![porque sim não é resposta!](http://www.eueleeascriancas.com.br/wp-content/uploads/2015/06/porque-sim.jpg)

Seguindo a logica bem empregada do @leocavalcante: 
> **Saber a pergunta certa é mais importante do que saber a resposta certa, pois toda solução deve estar alinhada com o problema que ele se propõe a resolver!**

Exemplo:
AngularJS 1.x ainda tem um número gigantesco de libs e conteúdo espalhado pela internet. E quando se fala em projetos rápidos que serão entregues a terceiros posteriormente para manutenção, AngularJS ainda é muito cotado pelos clientes, principalmente os que tem medo de sofrer uma rasteira do fenômeno _"qual o novo framework JS dessa semana?"_. Também existem outros cenários como: equipe, prazo, experiência, ... Mas acho já deu para pegar a ideia.

#### _"Nunca usei AngularJS 1.x, devo apostar nele para o futuro?"_

A resposta é **Sim** e **Não**.
Se você nunca brincou com ele, então você perdeu uma ótima experiência e oportunidade de entrar no clube dos caras que falam mal dele hoje(com propriedade é claro). Não existe experiência ruim, o ruim é não ter experiência! Então vá lá e tire o atraso.

Saber, ou ter vivido, as falhas de abordagens de desenvolvimento de grandes SPAs em AngularJS 1.x é tão ou mais importante do que saber novas abordagens com Redux, imutabilidade, funcional, ... Porque no final, quem te garante que você não está cometendo os mesmos erros de antigamente?

AngularJS foi um passo muito importante para a comunidade chegar onde chegou e só podemos vislumbrar um futuro melhor se soubermos como chegamos aqui. 

#jQueryIsAlive o/
